internal/provider/provider.go

```diff
@@ -231,6 +231,7 @@ func (p *MongodbtlasProvider) Configure(ctx context.Context, req provider.Config
 		PrivateKey:   data.PrivateKey.ValueString(),
 		BaseURL:      data.BaseURL.ValueString(),
 		RealmBaseURL: data.RealmBaseURL.ValueString(),
+		UserAgent:    config.TerraformVersionUserAgentInfo(req.TerraformVersion),
```

Comment:
it can be but I have kept it separate. Ideally user agent should be taken care of when creating new client (like we already do today) but in this case I had to set the new Terraform version user agent in the Provider because that's only where it's available.
Defining both in single place would mean we take provider version out of the client creation which I don't think we should do

Reply:
I see, one option could be to define a property TerraformVersion in the Config struct (instead of UserAgent), and then when creating a new client define the user agent with both the terraform and provider versions available. Let me know if you feel this makes it clearer or not.

Reply:
I was thinking the same thing after spending some time reviewing this. Having `TerraformVersion` would help a lot in terms of readability. In fact during client creation you can "compose" the user agent in-place without even having to have a special append function IMO. 

Also this solution would help to handle logs / problems: e.g. if `TerraformVersion` is empty during the client creation phase then we should handle it with a warning probably since it's something that needs attention

Reply:
I have added TerraformVersion in the Config struct, let me know what you think